pkg/client/restclient/config.go

```diff
@@ -158,6 +163,12 @@ func RESTClientFor(config *Config) (*RESTClient, error) {
 	if config.NegotiatedSerializer == nil {
 		return nil, fmt.Errorf("NegotiatedSerializer is required when initializing a RESTClient")
 	}
+	if config.QPS == 0.0 {
```

Comment:
I would suggest not to modify the config. Caller of `RESTClientFor` doesn't expect to get its config changed.

It seems the only place that uses config.QPS and config.Burst is `NewRESTClient`, could you copy them to two local variables and pass them to NewRESTClient?